extensions/bittrex/exchange.js

```diff
@@ -0,0 +1,289 @@
+          }
+          console.log(data.message)
+          return cb(null, [])
+        }
+
+        var balance = {
+          asset: 0,
+          currency: 0
+        }
+        Object.keys(data.result).forEach(function (i) {
+          var _balance = data.result[i]
+          if (_balance['Currency'] === opts.currency.toUpperCase()) {
+            balance.currency = n(_balance.Available).format('0.00000000'),
+              balance.currency_hold = 0
+          }
+          if (_balance['Currency'] === opts.asset.toUpperCase()) {
+            balance.asset = n(_balance.Available).format('0.00000000'),
+              balance.asset_hold = 0
+          }
+        })
+        cb(null, balance)
+      })
+    },
+
+    getQuote: function (opts, cb) {
+      var args = {
+        market: joinProduct(opts.product_id)
+      }
+      bittrex_public.getticker(args, function( data ) {
+        if (typeof data !== 'object') {
+          console.log('bittrex API (getticker) had an abnormal response, quitting.')
+          return cb(null, [])
+        }
+
+        if(!data.success) {
+          if (data.message.match(recoverableErrors)) {
+            return retry('getQuote', args, data.message)
+          }
+          console.log(data.message)
+          return cb(null, [])
+        }
+
+        cb(null, {
+          bid: data.result.Bid,
+          ask: data.result.Ask
+        })
+      })
+    },
+
+    cancelOrder: function (opts, cb) {
+      var args = [].slice.call(arguments)
+      bittrex_authed.cancel({
+        uuid: opts.order_id
+      }, function( data ) {
+        if (typeof data !== 'object') {
+          console.log('bittrex API (cancel) had an abnormal response, quitting.')
+          return cb(null, [])
+        }
+
+        if('error' in data || !data.success) {
+          console.log(data.error)
+          console.log(data.message)
+          return cb(null, [])
+        }
+
+        cb(null)
+      })
+    },
+
+    trade: function (type, opts, cb) {
+      var args = [].slice.call(arguments)
+
+      var params = {
+        market: joinProduct(opts.product_id),
+        quantity: opts.size,
+        rate: opts.price
+      }
+
+      if(!'ordertype' in opts) {
+        opts.ordertype = 'limit'
+      }
+
+      var fn = function(data) {
+        if (typeof data !== 'object') {
+          console.log('bittrex API (trade) had an abnormal response, quitting.')
+          return cb(null, [])
+        }
+
+        if(!data.success) {
+          if (data.message.match(recoverableErrors)) {
+            return retry('trade', args, data.message)
+          }
+          console.log(data.message)
+          return cb(null, [])
+        }
+
+        var order = {
+          id: data && data.result ? data.result.uuid : null,
+          status: 'open',
+          price: opts.price,
+          size: opts.size,
+          post_only: !!opts.post_only,
+          created_at: new Date().getTime(),
+          filled_size: '0',
```

Comment:
Maybe? I shamelessly copied this over from https://github.com/carlos8f/zenbot/blob/master/extensions/kraken/exchange.js#L200